python/lsst/analysis/tools/actions/plot/gridPlot.py

```diff
@@ -90,16 +103,49 @@ class GridPlot(PlotAction):
     def __call__(self, data: KeyedData, **kwargs) -> PlotResultType:
         """Plot data."""
         fig = plt.figure(figsize=self.figsize, dpi=self.dpi)
+        figureInfo = {"figsize": fig.get_size_inches(), "dpi": fig.get_dpi()}
+
+        if self.height_ratios is None:
+            height_ratios = np.ones(self.numRows) / self.numRows
+        else:
+            height_ratios = self.height_ratios / np.sum(self.height_ratios)
+
+        if self.width_ratios is None:
+            width_ratios = np.ones(self.numCols) / self.numCols
+        else:
+            width_ratios = self.width_ratios / np.sum(self.width_ratios)
+
+        includesPlotInfoPanel = False
+        addPlotInfoRow = 1
+        for panel in self.panels.values():
+            if isinstance(panel.plotElement, PlotInfoElement):
+                includesPlotInfoPanel = True
+                addPlotInfoRow = 0
+                break
+
+        if not includesPlotInfoPanel:
+            height_ratios = np.append([1.0], 20.0 * height_ratios)
+
         if self.suptitle is not None:
             fig.suptitle(**self.suptitle)
-        gs = GridSpec(self.numRows, self.numCols, figure=fig)
+        gs = GridSpec(
+            self.numRows + addPlotInfoRow,
+            self.numCols,
+            figure=fig,
+            height_ratios=height_ratios,
+            width_ratios=width_ratios,
+        )
+
+        if not includesPlotInfoPanel:
+            ax = fig.add_subplot(gs[0, :])
+            _ = PlotInfoElement().makeElement(ax=ax, figureInfo=figureInfo, **kwargs)
```

Comment:
Something about this doesn't feel quite right, i.e., a special carve-out for a `PlotInfoElement` (here and below). Perhaps the `data` for a `PlotInfoElement` is the `figureInfo`, which would then bring it into line with how other elements have been designed - i.e., "data input needed to make the element work".

Reply:
This may require us to re-think how we generate gridspecs somewhat though, i.e., we probably want the user to be able to specify the entire top-row in a gridplot should be reserved for use by a plot info element, rather than hard-coding it here.

Reply:
This "auto-add plotInfo" feature has been removed (see "in lieue" above).